src/lib/MnemonicPhrase.js

```diff
@@ -0,0 +1,199 @@
+class MnemonicPhrase {
+    /**
+     * @param {string} hex
+     * @return {Uint8Array}
+     */
+    static _hexToArray(hex) {
+        hex = hex.trim();
+        const bytes = hex.match(/.{2}/g) || [];
+        return Uint8Array.from(bytes.map(byte => parseInt(byte, 16)));
+    }
+
+    /**
+     * @param {*} buffer
+     * @return {string}
+     */
+    static _arrayToHex(buffer) {
+        let hex = '';
+        for (let i = 0; i < buffer.length; i++) {
+            const code = buffer[i];
+            hex += MnemonicPhrase.HEX_ALPHABET[code >>> 4];
+            hex += MnemonicPhrase.HEX_ALPHABET[code & 0x0F];
+        }
+        return hex;
+    }
+
+    // Adapted from https://github.com/mode80/crc8js
+    /**
+     *
+     * @param {Uint8Array} byte_array
+     */
+    static _crc8(byte_array) {
+        // TODO In the original, this table is generated each time. Intentionally?
+        let table = this._getCrcLookupTable();
+
+        // Calculate the 8-bit checksum given an array of byte-sized numbers
+        var c = 0;
+        for (var i = 0; i < byte_array.length; i++) {
+            c = table[(c ^ byte_array[i]) % 256];
+        }
+        return c;
+    }
+
+    static _getCrcLookupTable() {
+        // Create a lookup table byte array
+        if (!this._crcLookupTable) {
+            var table = []; // 256 max len byte array
+            for (var i = 0; i < 256; ++i) {
+                var curr = i;
+                for (var j = 0; j < 8; ++j) {
+                    if ((curr & 0x80) !== 0) {
+                        curr = ((curr << 1) ^ 0x97) % 256; // Polynomial C2 by Baicheva98
+                    } else {
+                        curr = (curr << 1) % 256;
+                    }
+                }
+                table[i] = curr;
+            }
+            this._crcLookupTable = table;
+        }
+        return this._crcLookupTable;
+    }
+
+    // Adapted from https://github.com/bitcoinjs/bip39
+    /**
+     *
+     * @param {string} str
+     * @param {string} padString
+     * @param {number} length
+     */
+    static _lpad(str, padString, length) {
+        while (str.length < length) str = padString + str;
+        return str;
+    }
+
+    /**
+     *
+     * @param {string} bin
+     */
+    static _binaryToBytes(bin) {
+        // TODO this method does not return number[] but just a single number... ?
```

Comment:
seems to be okay. It only gets called with binary strings that fit into a single number. However, you might add a check that Math.pow(2, bin.length) <= Number.MAX_SAFE_INTEGER